splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TimestampIT.java

```diff
@@ -0,0 +1,593 @@
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.splicemachine.derby.impl.sql.execute.operations;
+
+import com.splicemachine.derby.test.framework.SpliceSchemaWatcher;
+import com.splicemachine.derby.test.framework.SpliceUnitTest;
+import com.splicemachine.derby.test.framework.SpliceWatcher;
+import com.splicemachine.derby.test.framework.TestConnection;
+import com.splicemachine.homeless.TestUtils;
+import org.junit.*;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.junit.runners.Parameterized;
+import org.spark_project.guava.collect.Lists;
+
+import java.io.File;
+import java.sql.*;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test different operations on extended-range timestamps.
+ *
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(Parameterized.class)
+public class TimestampIT extends SpliceUnitTest {
+    private static final String SCHEMA = TimestampIT.class.getSimpleName().toUpperCase();
+    private Boolean useSpark;
+    private static boolean extendedTimestamps = true;
+    protected static SpliceWatcher spliceClassWatcher = new SpliceWatcher(SCHEMA);
+    protected static SpliceSchemaWatcher spliceSchemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    protected static SpliceWatcher methodWatcher = new SpliceWatcher(SCHEMA);
+
+    private static File BADDIR;
+
+    @ClassRule
+    public static TestRule chain = RuleChain.outerRule(spliceClassWatcher)
+                                            .around(spliceSchemaWatcher)
+                                            .around(methodWatcher);
+   // @Rule
+   // public SpliceWatcher methodWatcher = new SpliceWatcher(SCHEMA);
+     //    methodWatcher.setAutoCommit(false);
+
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = Lists.newArrayListWithCapacity(4);
+        params.add(new Object[]{true});
+        params.add(new Object[]{false});
+        return params;
+    }
+
+    public TimestampIT(Boolean useSpark) {
+        this.useSpark = useSpark;
+    }
+
+    @BeforeClass
+    public static void createDataSet() throws Exception {
+        methodWatcher.setAutoCommit(false);
+        createSharedTables(spliceClassWatcher.getOrCreateConnection());
+    }
+
+    public static void createSharedTables(Connection conn) throws Exception {
+        BADDIR = SpliceUnitTest.createBadLogDirectory(SCHEMA);
+        assertNotNull(BADDIR);
+
+        try (Statement s = conn.createStatement()) {
+            // s.execute("CALL SYSCS_UTIL.SYSCS_EMPTY_GLOBAL_STATEMENT_CACHE()");  Currently failing due to SPLICE-2217.
```

Comment:
FYI, with SPLICE-1814's fix, it seems that syscs_empty_global_statement_cache() is working now.